### ISSUE TITLE
Asset paths that begin with "/" don't work on Android

### ISSUE THREAD

**Issue by shadowislord** [jMonkeyEngine/jmonkeyengine#352]
Due to the changes in jME 3.1 to use Android's `AssetManager` instead of the `ClasspathLocator`, paths that begin with a forward slash no longer work.

**Comment by zzuegg** [comment]
Rather strange, there is already some code in the AndroidLocator to remove a leading /

**Comment by nordfalk** [comment]
OK, I have used several days on this one!

Here is what extensive debugging finally brought to me.
As you can see from https://github.com/jMonkeyEngine/jmonkeyengine/blob/master/jme3-android/src/main/resources/com/jme3/asset/Android.cfg the com.jme3.asset.plugins.AndroidLocator gets as / as path:

```
# Android specific locators
LOCATOR / com.jme3.asset.plugins.AndroidLocator
```

This means that the locator gets rootPath = "/" .... and now look at https://github.com/jMonkeyEngine/jmonkeyengine/blob/445f7ed010199d30c484fe75bacef4b87f2eb38e/jme3-android/src/main/java/com/jme3/asset/plugins/AndroidLocator.java#L26 , in the case where the AssetKey starts with a "/" (like "/Textures/...")

String assetPath = rootPath + key.getName(); // will start with TWO slashes, like "//Textures/...

The // Fix path issues will remove ONE of the slashes, but not the other, which will make the system look in the root of the Linux Android file system :-(


Proposed fix 1: Avoid / as path 

```
# Android specific locators
LOCATOR . com.jme3.asset.plugins.AndroidLocator
```
(best would be to set rootPath = "" but that isnt possible as it would not be valid syntax)

Proposed fix 2:  ... or change to a while loop instead of an if statement:

```
  // Fix path issues
        while (assetPath.startsWith("/")) {
            // Remove leading /
            assetPath = assetPath.substring(1);
        }
```

Proposed fix 3:  replace "//" with "/" before removing heading slash

```
        assetPath = assetPath.replace("//", "/");
        if (assetPath.startsWith("/")) {
            // Remove leading /
            assetPath = assetPath.substring(1);
        }

```


This bug affects all people that uses the SDK terrain editor (jMonkeyEngine 3.1-beta1-b002-SNAPSHOT) as it wrongly puths forward slashes in the paths of the textures in the .j3o file.

See also discussions
-  https://hub.jmonkeyengine.org/t/solved-jme-3-1-android-terrain-texture-problems/34274/25 (contains a workaround where all textures are set manually)
- https://hub.jmonkeyengine.org/t/google-cardboard-support/31684/33?u=jacobnordfalk  (issue 3)

**Comment by nordfalk** [comment]
Until this is fixed in the source the easiest workaround for the wrong path in the AndroidLocator is probably to unregister the locator with the wrong path and register it again with an empty path before you load the scene

```
        assetManager.unregisterLocator("/", AndroidLocator.class);
        assetManager.registerLocator("",AndroidLocator.class);

        Spatial scene = assetManager.loadModel("Scenes/myScene.j3o");

```

**Comment by stephengold** [comment]
Closed by PR #1577.
